abuu/classes/get_config_class.py

```diff
@@ -1,18 +1,28 @@
-from typing import Dict, Any
+from typing import Dict, Generic, TypeVar
 
-class ConfigClass:
-    pass
+V = TypeVar("V")
 
-"""
-A lot of config usually exists as JSON objects, when parsed in Python,
-it results in a dict being generated which can be accessed as dict['key']
+class ConfigClass(Generic[V]):
+    """
+    A lot of config usually exists as JSON objects, when parsed in Python,
+    it results in a dict being generated which can be accessed as dict['key']
 
-However, this looks less cleaner than writing config.key, which is more elegant.
-This function aims to convert a given config dict into a class which can be accessed as above.
-"""
-def get_config_class(config: Dict[str, Any]) -> ConfigClass:
-    config_class = ConfigClass()
-    for key, value in config.items():
-        assert isinstance(key, str), "The keys of config must be `str`. Invalid key: {}".format(key)
-        setattr(config_class, key, value)
-    return config_class
\ No newline at end of file
+    However, this looks less cleaner than writing config.key, which is more elegant.
+    This function aims to convert a given config dict into a class which can be accessed as above.
+    """
+    def __init__(self, data: Dict[str, V]) -> None:
```

Comment:
Wouldn't it be better to mark this as such since we don't know what the datatype of our values are, also all of them might not be the same datatype. 

We could possibly have it to be `Union[bool, str, int, float]`, but that would restrict it to being a config class for just JSON parsed dicts, however python dicts could have a lot more datatypes for values.

```suggestion
    def __init__(self, data: Dict[str, Any]) -> None:
```

Reply:
that is what using a TypeVar is doing, if the user passes a dict  that only contains strings than type checkers will know that the class will always return strings back. that is the magic of TypeVar and Generic

Reply:
Will this break if the dict's values have multiple datatypes?

Reply:
if the dict is type hinted with `Union[str, int]` for example, then type checkers will see `ConfigClass.dfdsnfosdf` as `Union[str, int]`.